ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -30,15 +30,31 @@
 
 #include <fstream>
 
+namespace gcs = google::cloud::storage;
+
 gcp_connection::gcp_connection(const std::string &bucket_name)
     : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name)
 {
 }
 
 // Builds a list of object names from the bucket.
 int
-gcp_connection::list_objects(std::vector<std::string> &objects) const
+gcp_connection::list_objects(std::vector<std::string> &objects, bool list_single) const
 {
+    using namespace gcs;
+    gcs::Client client = _gcp_client;
```

Comment:
Ideally we want to remove this and directly use _gcp_client, I remember we had a const compiler error can we fix that?

Reply:
We are unable to directly use _gcp_client with the ListObjects() method, it's giving an error that says argument type does not match when I give it a argument of the correct type. So I'm using a seperate variable 'client' for now, I can do some investigation on that and see if there is a way I can directly use _gcp_client.

Reply:
Let's fix this since we had the conversation with sona.